modules/pico_dev_ppp.c

```diff
@@ -302,6 +302,10 @@ static void lcp_optflags_print(struct pico_device_ppp *ppp, uint8_t *opts, uint3
 #define PPP_TIMER_ON_AUTH       0x10u
 #define PPP_TIMER_ON_IPCP       0x20u
 
+static int should_escape(uint8_t byte) {
+  return (byte == PPPF_FLAG_SEQ || byte == PPPF_CTRL_ESC || byte < 0x20);
+}
+
```

Comment:
This PPP stuff doesn't really belong in this PR I think, but I must admit that it's much nicer code :-)
But unless I'm mistaken, the `byte < 0x20` part isn't equivalent with the old code - what's your reasoning behind this change? (Not a huge expert on PPP myself)